kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/BigQuerySinkConfig.java

```diff
@@ -645,373 +647,382 @@ private void logDeprecationWarnings() {
+            ).define(
+                    THREAD_POOL_SIZE_CONFIG,
+                    THREAD_POOL_SIZE_TYPE,
+                    THREAD_POOL_SIZE_DEFAULT,
+                    THREAD_POOL_SIZE_VALIDATOR,
+                    THREAD_POOL_SIZE_IMPORTANCE,
+                    THREAD_POOL_SIZE_DOC
+            ).define(
+                    QUEUE_SIZE_CONFIG,
+                    QUEUE_SIZE_TYPE,
+                    QUEUE_SIZE_DEFAULT,
+                    QUEUE_SIZE_VALIDATOR,
+                    QUEUE_SIZE_IMPORTANCE,
+                    QUEUE_SIZE_DOC
+            ).define(
+                    BIGQUERY_RETRY_CONFIG,
+                    BIGQUERY_RETRY_TYPE,
+                    BIGQUERY_RETRY_DEFAULT,
+                    BIGQUERY_RETRY_VALIDATOR,
+                    BIGQUERY_RETRY_IMPORTANCE,
+                    BIGQUERY_RETRY_DOC
+            ).define(
+                    BIGQUERY_RETRY_WAIT_CONFIG,
+                    BIGQUERY_RETRY_WAIT_CONFIG_TYPE,
+                    BIGQUERY_RETRY_WAIT_DEFAULT,
+                    BIGQUERY_RETRY_WAIT_VALIDATOR,
+                    BIGQUERY_RETRY_WAIT_IMPORTANCE,
+                    BIGQUERY_RETRY_WAIT_DOC
+            ).define(
+                    BIGQUERY_MESSAGE_TIME_PARTITIONING_CONFIG,
+                    BIGQUERY_MESSAGE_TIME_PARTITIONING_CONFIG_TYPE,
+                    BIGQUERY_MESSAGE_TIME_PARTITIONING_DEFAULT,
+                    BIGQUERY_MESSAGE_TIME_PARTITIONING_IMPORTANCE,
+                    BIGQUERY_MESSAGE_TIME_PARTITIONING_DOC
+            ).define(
+                    BIGQUERY_PARTITION_DECORATOR_CONFIG,
+                    BIGQUERY_PARTITION_DECORATOR_CONFIG_TYPE,
+                    BIGQUERY_PARTITION_DECORATOR_DEFAULT,
+                    BIGQUERY_PARTITION_DECORATOR_IMPORTANCE,
+                    BIGQUERY_PARTITION_DECORATOR_DOC
+            ).define(
+                    BIGQUERY_TIMESTAMP_PARTITION_FIELD_NAME_CONFIG,
+                    BIGQUERY_TIMESTAMP_PARTITION_FIELD_NAME_TYPE,
+                    BIGQUERY_TIMESTAMP_PARTITION_FIELD_NAME_DEFAULT,
+                    BIGQUERY_TIMESTAMP_PARTITION_FIELD_NAME_VALIDATOR,
+                    BIGQUERY_TIMESTAMP_PARTITION_FIELD_NAME_IMPORTANCE,
+                    BIGQUERY_TIMESTAMP_PARTITION_FIELD_NAME_DOC
+            ).define(
+                    BIGQUERY_CLUSTERING_FIELD_NAMES_CONFIG,
+                    BIGQUERY_CLUSTERING_FIELD_NAMES_TYPE,
+                    BIGQUERY_CLUSTERING_FIELD_NAMES_DEFAULT,
+                    BIGQUERY_CLUSTERING_FIELD_NAMES_VALIDATOR,
+                    BIGQUERY_CLUSTERING_FIELD_NAMES_IMPORTANCE,
+                    BIGQUERY_CLUSTERING_FIELD_NAMES_DOC
+            ).define(
+                    new ConfigKeyBuilder<>(TIME_PARTITIONING_TYPE_CONFIG)
+                            .type(TIME_PARTITIONING_TYPE_TYPE)
+                            .defaultValue(TIME_PARTITIONING_TYPE_DEFAULT)
+                            .validator(
```

Comment:
Huh, as an API improvement, what do you think about going BACK to the lambda style parameter (from before #102) and passing an optional second parameter for the `toString()` document string?   Not very important but it would clear out some extra boilerplate from _this_ part of the code (although admittedly moving it there).

Not a blocker, just an idea!